alfred-telemetry-platform/src/main/java/eu/xenit/alfred/telemetry/binder/solr/sharding/SolrShardingMetricsFactory.java

```diff
@@ -26,6 +31,12 @@ public SolrShardingMetricsFactory(ShardRegistry shardRegistry, MeterRegistry reg
                 .forJob(jobDetail.getKey()).build();
 
         scheduler.scheduleJob(jobDetail, trigger);
+        this.jobDetail = jobDetail;
+        this.trigger = trigger;
+    }
+
+    public void destroy() throws SchedulerException {
```

Comment:
In what cases can the `SchedulerException` be thrown?

I don't know for sure, but throwing from the destroy method may result in Spring shutting down the whole application.
In that case, a reload attempt of a solr subsystem would result in Alfresco becoming unavailable.

It may be better to put a try-catch around it and log the error. If we can't unschedule the job, then at least we won't shutdown Alfresco for this.